File: src/main/java/org/apache/fineract/client/services/BatchApiApi.java
/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements. See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership. The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
package org.apache.fineract.client.services;

import org.apache.fineract.client.CollectionFormats.*;

import rx.Observable;
import retrofit2.http.*;

import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import okhttp3.MultipartBody;

import org.apache.fineract.client.models.BatchResponse;
import org.apache.fineract.client.models.PostBatchesRequest;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public interface BatchApiApi {
  /**
   * Batch requests in a single transaction
   * The Apache Fineract Batch API is also capable of executing all the requests in a single transaction, by setting a Query Parameter, \&quot;enclosingTransaction&#x3D;true\&quot;. So, if one or more of the requests in a batch returns an erroneous response all of the Data base transactions made by other successful requests will be rolled back.  If there has been a rollback in a transaction then a single response will be provided, with a &#39;400&#39; status code and a body consisting of the error details of the first failed request.
   * @param postBatchesRequest  (required)
   * @param enclosingTransaction enclosingTransaction (optional, default to false)
   * @return Observable&lt;BatchResponse&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @POST("batches")
  Observable<BatchResponse> handleBatchRequests(
    @retrofit2.http.Body PostBatchesRequest postBatchesRequest, @retrofit2.http.Query("enclosingTransaction") Boolean enclosingTransaction
  );

}

File: src/main/kotlin/org/openapitools/client/apis/BatchAPIApi.kt
package org.openapitools.client.apis

import org.openapitools.client.models.BatchRequest
import org.openapitools.client.models.BatchResponse
import retrofit2.http.Body
import retrofit2.http.POST
import retrofit2.http.Query

interface BatchAPIApi {
    /**
     * Batch requests in a single transaction
     * The Apache Fineract Batch API is also capable of executing all the requests in a single transaction, by setting a Query Parameter, \&quot;enclosingTransaction&#x3D;true\&quot;. So, if one or more of the requests in a batch returns an erroneous response all of the Data base transactions made by other successful requests will be rolled back.  If there has been a rollback in a transaction then a single response will be provided, with a &#39;400&#39; status code and a body consisting of the error details of the first failed request.
     * Responses:
     *  - 200: Success
     *
     * @param batchRequest
     * @param enclosingTransaction enclosingTransaction (optional, default to false)
     * @return [kotlin.collections.List<BatchResponse>]
     */
    @POST("v1/batches")
    suspend fun handleBatchRequests(
        @Body batchRequest: List<BatchRequest>,
        @Query("enclosingTransaction") enclosingTransaction: Boolean? = false
    ): List<BatchResponse>

}
